packages/react-instantsearch-hooks-dom/src/ui/lib/cx.ts

```diff
@@ -1,5 +1,9 @@
-export function cx(
-  ...classNames: Array<string | number | boolean | undefined | null>
-) {
-  return classNames.filter(Boolean).join(' ');
+type Root = string | number | boolean | undefined | null;
+export type CSSClass = Root | Root[];
+
+export function cx(...classNames: CSSClass[]) {
+  return classNames
+    .reduce<Root[]>((acc, curr) => acc.concat(curr), [])
+    .filter(Boolean)
```

Comment:
If we keep this (see previous comment) could we do the filtering in `reduce` to avoid the two passes? And test it?

Reply:
If you'd pass in reduce immediately, you'd need something like `acc.concat(Array.isArray(curr) ? curr.filter(Boolean) : [curr].filter(Boolean))` or similar, that doesn't seem better to me

Reply:
Fair, and the performance implications are likely nonexistent.